cognite/client/_api/files.py

```diff
@@ -525,6 +534,8 @@ def upload(
         if os.path.isfile(path):
             if not name:
                 file_metadata.name = os.path.basename(path)
+            if not directory:
+                file_metadata.directory = os.path.dirname(path)
```

Comment:
Are we sure that we want to fill this in if the user does not provide it?

Reply:
I see now that this is in the `upload` method, which uploads from a file on disk.

Still, I'm not sure we want to do this. The file name is nice to provide, but the user might not expect that the SDK reads and fills in the directory.

Reply:
Yeah I wasn't really sure about this. On one hand it's really convenient *when* the user wants it but as you say, they might not always want that to happen. And since this is the absolute path it might also contain some personal-_ish_ data, like their computer name. 

So perhaps it's better to just not do this at all?

Reply:
I agree! I vote for no! But I'd be interested to hear others chime in as well

Reply:
I'm also going to vote for no. Not only could we hit personal data problem but the original file directory might not match the upload directory

Reply:
+1 for not doing this